includes/wc-user-functions.php

```diff
@@ -803,3 +803,21 @@ function wc_update_user_last_active( $user_id ) {
 	}
 	update_user_meta( $user_id, 'wc_last_active', (string) strtotime( date( 'Y-m-d', current_time( 'timestamp', true ) ) ) );
 }
+
+/**
+ * Translate WC roles using the woocommerce textdomain.
+ *
+ * @since 3.7.0
+ * @param string $translation  Translated text.
+ * @param string $text         Text to translate.
+ * @param string $context      Context information for the translators.
+ * @param string $domain       Text domain. Unique identifier for retrieving translated strings.
+ * @return string
+ */
+function wc_translate_user_roles( $translation, $text, $context, $domain ) {
+	if ( 'User role' === $context && 'default' === $domain && in_array( $text, array( 'Shop manager', 'Customer' ), true ) ) {
+		return translate_user_role( $text, 'woocommerce' );
```

Comment:
`translate_user_role` accepts only one argument.
https://developer.wordpress.org/reference/functions/translate_user_role/

Also this is generating a infinity loop:

```
[02-Apr-2019 18:40:45 UTC] PHP Fatal error:  Allowed memory size of 268435456 bytes exhausted (tried to allocate 262144 bytes) in wp-includes/l10n.php on line 197
[02-Apr-2019 18:40:45 UTC] PHP Fatal error:  Allowed memory size of 268435456 bytes exhausted (tried to allocate 262144 bytes) in wp-includes/load.php on line 788
```

I managed to fix it with:

```php
function wc_translate_user_roles( $translation, $text, $context, $domain ) {
	if ( 'User role' === $context && 'default' === $domain && in_array( $text, array( 'Shop manager', 'Customer' ), true ) ) {
		if ( 'Shop manager' === $text ) {
			return _x( 'Shop manager', 'User role', 'woocommerce' );
		} elseif ( 'Customer' === $text ) {
			return _x( 'Customer', 'User role', 'woocommerce' );
		}
	}

	return $translation;
}
add_filter( 'gettext_with_context', 'wc_translate_user_roles', 10, 4 );
```

We could use `_x( $text, 'User role', 'woocommerce' );`, for sure less code, but this can generate problems with indexing translatable strings (on GlotPress (translations.wordpress.org), a few plugins like Loco translate and so on).

Reply:
Looks like this was added in 5.2 https://github.com/WordPress/WordPress/blob/24e2829eb95e2fdb31c5774b510d0b39cee13034/wp-includes/l10n.php#L1242

Reply:
I tested with 5.2, still generating a infinity loop.

Reply:
Works just fine, seems like some problem with the installed version.
I include some code to avoid load it when not using WP 5.2.